cibyl/cli/parser.py

```diff
@@ -99,6 +101,21 @@ def print_help(self) -> None:
         arguments that are currently added."""
         self.argument_parser.print_help()
 
+    def add_argument_to_tree(self, arg: Argument,
+                             parent_queries: Set[str]) -> None:
+        """Add argument to the tree of query method relationships. This tree
+        relates query method that have dependencies, e.g. get_jobs and
+        get_builds. Queries are added as nodes and the edges of the graph marks
+        the relationships. Nodes cannot be duplicated and self-links are
+        avoided."""
+        if arg.func in parent_queries:
+            # avoid creating self-links
+            return
+        self.graph_queries.add_node(arg.func)
+        for query in parent_queries:
+            self.graph_queries.add_node(query)
```

Comment:
I gather that if the node already exists, this will do nothing then.

Reply:
Yes, that's right, I'll specify it in a comment since it's a feature of the networkx library, thanks for pointing it out

Reply:
Great.